internal/controller/correlation/middleware.go

```diff
@@ -32,18 +34,20 @@ func OnResponseComplete(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		begin := time.Now()
 		next.ServeHTTP(w, r)
-		correlationId := FromContext(r.Context())
-		if common.LoggingClient != nil {
-			common.LoggingClient.Trace("Response complete", clients.CorrelationHeader, correlationId, "duration", time.Since(begin).String())
+		correlationId := IdFromContext(r.Context())
+		lc := LoggingClientFromContext(r.Context())
+		if lc != nil {
+			LoggingClient.Trace("Response complete", clients.CorrelationHeader, correlationId, "duration", time.Since(begin).String())
 		}
 	})
 }
 
 func OnRequestBegin(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		correlationId := FromContext(r.Context())
-		if common.LoggingClient != nil {
-			common.LoggingClient.Trace("Begin request", clients.CorrelationHeader, correlationId, "path", r.URL.Path)
+		correlationId := IdFromContext(r.Context())
+		lc := LoggingClientFromContext(r.Context())
+		if lc != nil {
+			LoggingClient.Trace("Begin request", clients.CorrelationHeader, correlationId, "path", r.URL.Path)
```

Comment:
I would prefer to create a  local in line logger instance in the rare case `lc` is nil so we can remove the Global instance.